### PR TITLE
Announce 4.103

### PR DESCRIPTION
**Pull request by fredemmott**

### CODE REVIEW

**Comment by fredemmott** [comment]
Builds are still in progress: delayed by macos issues

File: _posts/2021-03-31-hhvm-4.103.markdown [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+HHVM 4.103 is released! This release marks the end of support for 4.96;
+HHVM 4.98&ndash;4.101 remain supported, as do the 4.56, 4.80, and 4.102 LTS
+releases.
+
+# Highlights
+
+- functions with implicit/no return are now considered a subtype of functions
+  that return `mixed` or `dynamic`; for example, it is now possible to pass
+  `() ==> {echo "hello;" }` as a `(function(): mixed)`
```
Comment (lexidor) [inline _posts/2021-03-31-hhvm-4.103.markdown:16]: The `;` needs to be moved to outside the string literal.

File: _posts/2021-03-31-hhvm-4.103.markdown [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+title: "HHVM 4.103"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.103 is released! This release marks the end of support for 4.96;
```
Comment (jjergus) [inline _posts/2021-03-31-hhvm-4.103.markdown:8]:
```suggestion
HHVM 4.103 is released! This release marks the end of support for 4.97;
```

File: _posts/2021-03-31-hhvm-4.103.markdown [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- Hack arrays and legacy arrays are unified; due to the potential scope of this
+  change, 4.102 [has been made](https://hhvm.com/blog/2021/03/29/extending-hhvm-4.102-support.html)
+  an additional out-of-schedule LTS release to allow more time to determine 
+  migration paths.
+  - `varray` is now an alias for `vec`
+  - `darray` is now an alias for `dict`
+  - shapes are now backed by `dict`s at runtime; this generally remains
+    considered an implementation detail, but will be helpful when handling
+    untyped data such as decoded JSON.
+  - tuples are now backed by `vec`s at runtime; this is also an considered
```
Comment (jjergus) [inline _posts/2021-03-31-hhvm-4.103.markdown:37]:
```suggestion
  - tuples are now backed by `vec`s at runtime; this is also considered
```

File: _posts/2021-03-31-hhvm-4.103.markdown [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  constants; this is ongoing work, and we expect more circular definitions to
+  be detected in future versions.
+- `apc_prime.so` support has been removed; this was an undocumented feature that
+  would allow providing a C++ library to pre-fill APC on HHVM startup.
+- a typechecker error is now raised if an XHP class declares two attributes with
+  the same name.
+- PowerPC64 support has been removed; it has been unusable for some time, and we
+  do not have the CI infrastructure necessary to support it.
+
+# Future Changes
```
Comment (jjergus) [inline _posts/2021-03-31-hhvm-4.103.markdown:52]: Maybe also list as future change: Support for MacOS 10.14 (Mojave) will likely be dropped soon.

**Review by jjergus — APPROVED** [review]
style nit: I think preferably bullet points should either be sentences (start with capital letter) or not end with periods (or at least be consistent with each other) though I will understand if you don't want to waste time with this :)

File: _posts/2021-03-31-hhvm-4.103.markdown (hunk 2 above)
Comment (fredemmott) [reply _posts/2021-03-31-hhvm-4.103.markdown:8]: Hmm, how come? 4.97 is 6 weeks old, wasn't an LTS, and 4.102 also said 4.97

Comment (jjergus) [reply _posts/2021-03-31-hhvm-4.103.markdown:8]: 4.102 announcement says "end of support for 4.96", that's why I thought this should be 97

There should be 6 supported releases (+any older LTS), so we have 98, 99, 100, 101, 102, 103 now

Comment (fredemmott) [reply _posts/2021-03-31-hhvm-4.103.markdown:8]: I misread, I thought you were suggesting the opposite to this change :)

**Comment by fredemmott** [comment]
Debian 10 buster build appears to have an AWS flakiness issue; re-ran, hopefully done in a few more hours.

**Comment by fredemmott** [comment]
Pushed this yesterday, surprised this wasn't closed :/

**Pull request closed without merging**